marklogic-data-hub/src/main/java/com/marklogic/hub/impl/MasteringManagerImpl.java

```diff
@@ -114,9 +128,48 @@ public MergeResponse merge(List<String> mergeURIs, String flowName, String stepN
                     resp = null;
                 } else {
                     ResourceServices.ServiceResult res = resultItr.next();
-                    StringHandle handle = new StringHandle();
-                    ObjectMapper objectMapper = new ObjectMapper();
-                    resp = objectMapper.readValue(res.getContent(handle).get(), MergeResponse.class);
+                    JacksonHandle handle = new JacksonHandle();
+                    resp = res.getContent(handle).get();
+                }
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            } finally {
+                if (resultItr != null) {
+                    resultItr.close();
+                }
+            }
+            return resp;
+        }
+    }
+
+    static class MatchResource extends ResourceManager {
+        private String targetDatabase;
+
+        public MatchResource(DatabaseClient srcClient, String targetDatabase) {
+            super();
+            this.targetDatabase = targetDatabase;
+            srcClient.init("mlSmMatch" , this);
+        }
+
+        public JsonNode match(String matchURI, String flowName, String stepNumber, Boolean includeMatchDetails, JsonNode options) {
+            JsonNode resp;
+
+            RequestParameters params = new RequestParameters();
+            params.put("uri", matchURI);
+            params.put("includeMatchDetails", includeMatchDetails.toString());
+            params.put("flowName", flowName);
+            params.put("step", stepNumber);
+            params.put("targetDatabase", targetDatabase);
+            params.put("sourceDatabase", targetDatabase);
+            JacksonHandle jsonOptions = new JacksonHandle().with(options);
+            ResourceServices.ServiceResultIterator resultItr = this.getServices().post(params, jsonOptions);
+            try {
+                if (resultItr == null || !resultItr.hasNext()) {
```

Comment:
Same thing, I think you just need post(params, jsonOptions, new JacksonHandle()).get()